spec/requests/posts_spec.rb

```diff
@@ -1,7 +1,46 @@
 require 'rails_helper'
 
 RSpec.describe 'Posts', type: :request do
+  before(:all) do
+    @user = User.new(name: 'Bianca')
+    @user.save
+  end
+
   describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+    before do
+      get user_posts_path(@user)
+    end
+
+    it 'response status correct' do
+      expect(response.status).to eq(200)
+      expect(response).to be_successful
+    end
+
+    it 'renders the index template' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'response body includes correct placeholders text' do
+      expect(response.body).to include('Posts')
+    end
+  end
+
+  describe 'GET /show' do
+    before do
+      @post = Post.create(author: @user, title: 'Test Post')
+      get user_post_path(@user, @post)
+    end
+
+    it 'response status correct' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the show template' do
+      expect(response).to render_template(:show)
+    end
+
+    it 'response body includes correct placeholders text' do
+      expect(response.body).to include('Post selected Info')
+    end
```

Comment:
- [ ] Please 🙏🏽 correct the test for the correct TDD 

```ruby
1) Posts GET /index renders the index template
     Failure/Error: expect(response).to render_template(:index)
     
     NoMethodError:
       assert_template has been extracted to a gem. To continue using it,
               add `gem 'rails-controller-testing'` to your Gemfile.
     # ./spec/requests/posts_spec.rb:20:in `block (3 levels) in <main>'

  2) Posts GET /show renders the show template
     Failure/Error: expect(response).to render_template(:show)
     
     NoMethodError:
       assert_template has been extracted to a gem. To continue using it,
               add `gem 'rails-controller-testing'` to your Gemfile.
     # ./spec/requests/posts_spec.rb:39:in `block (3 levels) in <main>'
```

Reply:
@franclobo  Can you help me with the problem? I can't fix the problem when I try to use render_template.
Can you guide me to solve it? Thank you